packages/components/Menu/src/MenuItem/useMenuItem.ts

```diff
@@ -60,11 +61,25 @@ export const useMenuItem = (props: MenuItemProps): MenuItemInfo => {
   // Explicitly override onKeyDown to override the native behavior of moving focus with arrow keys.
   const onKeyDownProps = useKeyDownProps(onInvoke, ...keys);
 
-  useHoverFocusEffect(pressable.state.hovered, componentRef);
+  useHoverFocusEffect(pressable.state.hovered, componentRef, setIsMenuItemHovered);
+  useRemoveHoverEffect(pressable.state.hovered, menuListContext.isMenuItemHovered, pressable.props.onHoverOut);
+
+  const onFocusOverride = React.useCallback(
+    (e) => {
+      pressable.props.onFocus(e);
+      // if it's not a hover focus
+      if (!pressable.state.focused && !pressable.state.hovered) {
+        // remove hover state from everywhere else
```

Comment:
I'm guessing that onBlur doesn't fire on win32? Otherwise that seems like a better solution (listen for when focus moves somewhere else and clear hover in that case)